src/compiler/parser.ts

```diff
@@ -584,6 +591,20 @@ namespace ts {
             return result;
         }
 
+        export function parseIsolatedEntityName(content: string, allowReservedWords = false) {
+            initializeState("file.js", content, ScriptTarget.Latest, /*_syntaxCursor:*/ undefined, ScriptKind.JS);
+            sourceFile = <SourceFile>{ languageVariant: LanguageVariant.Standard, text: content };
+
+            nextToken();
+            const entityName = parseEntityName(allowReservedWords);
+            parseExpected(SyntaxKind.EndOfFileToken);
+
+            const diagnostics = parseDiagnostics;

```

Comment:
Are you using these diagnostics, other than to check that they exist in program.ts:1546? You could return `undefined` rather than allocating the object literal for every call.